### PR TITLE
Modify how belts and rings are differentiated in JournalScan.cs

### PR DESCRIPTION
**Pull request by spadino**
Very simple modification to allow for extremely rare ringed stars.

### CODE REVIEW

File: EDDiscovery/UserControls/UserControlScanGrid.cs [hunk 1]
```diff
@@ -195,22 +195,53 @@ void DrawSystem(HistoryEntry he, bool force)
                         bdDetails.Append("Volcanism".Tx(this) + ". ");
 
                     // have some ring?
-                    if (sn.ScanData.HasRings && sn.ScanData.IsStar == false)
+                    if (sn.ScanData.HasBelts || sn.ScanData.HasRings)
```
Comment (klightspeed) [inline EDDiscovery/UserControls/UserControlScanGrid.cs]: HasBelts seems to be superfluous, and is not initialized until `DisplayString` is called.

File: EDDiscovery/UserControls/UserControlScanGrid.cs [hunk 2]
```diff
@@ -195,22 +195,53 @@ void DrawSystem(HistoryEntry he, bool force)
...
+                            {
+                                bdDetails.Append(string.Format("Has {0} rings: ".Tx(this, "Rings"),
+                                                               sn.ScanData.Rings.Count()));
+                            }
+                            else
+                            {
+                                bdDetails.Append("Has 1 ring: ".Tx(this, "Ring"));
+                            }
+
+                            foreach (var r in sn.ScanData.Rings)
```
Comment (klightspeed) [inline EDDiscovery/UserControls/UserControlScanGrid.cs]: Should this perhaps be outside the `if (sn.ScanData.Rings[0].Name.EndsWith("{Belt,Ring}"))` - it is identical to foreach block in the Belts block.

File: EliteDangerous/JournalEvents/JournalScan.cs [hunk 3]
```diff
@@ -480,28 +483,33 @@ public string DisplayString(int indent = 0, bool includefront = true , MaterialC
...
-                if (IsStar)
+                for (var r = 0; r < Rings.Length; r++)
                 {
-                    scanText.AppendFormat("Belt{0}".Tx(this), Rings.Count() == 1 ? ":" : "s:");
-                    for (int i = 0; i < Rings.Length; i++)
+                    // If it's name ends with "Belt"...
+                    if (Rings[r].Name.EndsWith("Belt"))
                     {
-                        if (Rings[i].MassMT > (oneMoon_MT / 10000))
+                        HasBelts = true;
```
Comment (klightspeed) [inline EliteDangerous/JournalEvents/JournalScan.cs]: Such variables should probably be initialized inside the constructor.

File: EliteDangerous/JournalEvents/JournalScan.cs [hunk 4]
```diff
@@ -480,28 +483,33 @@ public string DisplayString(int indent = 0, bool includefront = true , MaterialC
...
+
+                        // ...than it's a belt!
+                        scanText.AppendFormat("Belt{0}".Tx(this), Rings.Count() == 1 ? ":" : "s:");
+
+                        for (r = 0; r < Rings.Length; r++)
                         {
-                            scanText.Append("\n" + RingInformation(i, 1.0 / oneMoon_MT, " Moons".Tx(this)));
+                            scanText.Append("\n" + RingInformation(r));
```
Comment (klightspeed) [inline EliteDangerous/JournalEvents/JournalScan.cs]: This removes the mass scaling

File: EliteDangerous/JournalEvents/JournalScan.cs [hunk 5]
```diff
@@ -480,28 +483,33 @@ public string DisplayString(int indent = 0, bool includefront = true , MaterialC
...
-                    scanText.AppendFormat("Belt{0}".Tx(this), Rings.Count() == 1 ? ":" : "s:");
-                    for (int i = 0; i < Rings.Length; i++)
+                    // If it's name ends with "Belt"...
+                    if (Rings[r].Name.EndsWith("Belt"))
                     {
-                        if (Rings[i].MassMT > (oneMoon_MT / 10000))
+                        HasBelts = true;
+
+                        // ...than it's a belt!
+                        scanText.AppendFormat("Belt{0}".Tx(this), Rings.Count() == 1 ? ":" : "s:");
```
Comment (klightspeed) [inline EliteDangerous/JournalEvents/JournalScan.cs]: Probably need the Belt(s) and Ring(s) to have separate translated singular and plural forms.

File: EDDiscovery/UserControls/UserControlScanGrid.cs (hunk 1 above)
Comment (spadino) [reply EDDiscovery/UserControls/UserControlScanGrid.cs]: Yes, it's a placeholder for further improvements.

File: EliteDangerous/JournalEvents/JournalScan.cs (hunk 5 above)
Comment (spadino) [reply EliteDangerous/JournalEvents/JournalScan.cs]: Yeah, good point...

File: EliteDangerous/JournalEvents/JournalScan.cs (hunk 4 above)
Comment (spadino) [reply EliteDangerous/JournalEvents/JournalScan.cs]: Ops, unintended deletion...

File: EDDiscovery/UserControls/UserControlScanGrid.cs (hunk 2 above)
Comment (spadino) [reply EDDiscovery/UserControls/UserControlScanGrid.cs]: I have implemented some different solution, that one seemed simple enough. Anyway, I'll wait for the next updates to rework the whole idea. Thanks!

**Comment by spadino** [comment]
@klightspeed, thanks for the comments! I'll rework the idea.

**Pull request closed without merging**